### PR TITLE
docs: remove hyperOpt exclusion - mlflow on synapse

### PR DESCRIPTION
**Pull request by JessicaXYWang**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

_Briefly describe the changes included in this Pull Request._

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

### CODE REVIEW

File: notebooks/features/hyperparameter_tuning/HyperOpt-SynapseML.ipynb [hunk 1]
```diff
@@ -29,7 +29,10 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%pip install hyperopt"
+    "# %pip install hyperopt\n",
+    "import os\n",
+    "\n",
+    "os.system(\"pip install hyperopt --no-cache-dir --force-reinstall\")"
```
Comment (serena-ruan) [inline notebooks/features/hyperparameter_tuning/HyperOpt-SynapseML.ipynb:35]:
```suggestion
    "os.system(\"pip install hyperopt\")"
```

File: notebooks/features/hyperparameter_tuning/HyperOpt-SynapseML.ipynb [hunk 2]
```diff
@@ -49,7 +52,31 @@
    "outputs": [],
    "source": [
     "# version >= 1.28.0 supports reading logModelAllowlistFile from url\n",
-    "%pip install mlflow==1.29.0"
+    "# %pip install mlflow==1.29.0\n",
+    "os.system(\"pip install mlflow==1.29.0 --no-cache-dir --force-reinstall\")"
```
Comment (serena-ruan) [inline notebooks/features/hyperparameter_tuning/HyperOpt-SynapseML.ipynb:56]:
```suggestion
    "os.system(\"pip install mlflow==1.29.0\")"
```
We don't need the extra config if the package is not installed already.

File: notebooks/features/hyperparameter_tuning/HyperOpt-SynapseML.ipynb [hunk 3]
```diff
@@ -100,14 +127,13 @@
...
     "\n",
-    "if running_on_synapse_internal():\n",
-    "    experiment_name = \"hyperopt-synapseml\"\n",
-    "else:\n",
-    "    experiment_name = \"/Shared/hyperopt-synapseml\"\n",
+    "if running_on_synapse():\n",
+    "    from notebookutils.mssparkutils import azureML\n",
     "\n",
-    "# Set MLflow experiment.\n",
+    "    ws = azureML.getWorkspace(\"AzureMLService1\")\n",
```
Comment (mhamilton723) [inline notebooks/features/hyperparameter_tuning/HyperOpt-SynapseML.ipynb:135]: Likewise here could you pull this into the top cell called prerequisites where you show people how to set this up on synapse

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/mlflow/installation.md](https://github.com/microsoft/SynapseML/blob/f745c7896fe20829e4a117c19dcbdb802c608067/website%2Fdocs%2Fmlflow%2Finstallation.md) | 79 | 100 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/e320027b-3398-4d70-bb55-301ad962c05e) | [:white_check_mark:](# "Successfully processed") |
| [website/docs/mlflow/introduction.md](https://github.com/microsoft/SynapseML/blob/f745c7896fe20829e4a117c19dcbdb802c608067/website%2Fdocs%2Fmlflow%2Fintroduction.md) | 88 | 100 | 71 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/39aebf70-a099-4b2f-b62f-a1189d55fffb) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/mlflow/installation.md](https://github.com/microsoft/SynapseML/blob/0b42f879e9ffa9f1dc1caba249540ff3eb4e0a34/website%2Fdocs%2Fmlflow%2Finstallation.md) | 78 | 100 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/c5a56c96-4c2c-4630-82c4-5c0ae0ff67b2) | [:white_check_mark:](# "Successfully processed") |
| [website/docs/mlflow/introduction.md](https://github.com/microsoft/SynapseML/blob/0b42f879e9ffa9f1dc1caba249540ff3eb4e0a34/website%2Fdocs%2Fmlflow%2Fintroduction.md) | 88 | 100 | 71 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/49df74db-a6eb-4cc0-b129-a855b2e85917) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/mlflow/installation.md](https://github.com/microsoft/SynapseML/blob/de7a7b62cff891cc6083e60c35f02735527624c4/website%2Fdocs%2Fmlflow%2Finstallation.md) | 78 | 100 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/b7a43771-a497-40ad-8d3e-88271c7e12b6) | [:white_check_mark:](# "Successfully processed") |
| [website/docs/mlflow/introduction.md](https://github.com/microsoft/SynapseML/blob/de7a7b62cff891cc6083e60c35f02735527624c4/website%2Fdocs%2Fmlflow%2Fintroduction.md) | 88 | 100 | 71 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/2ec49701-5d46-42f2-ba5a-67db32ab380d) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/mlflow/installation.md](https://github.com/microsoft/SynapseML/blob/42e0eaa90579b62d3277a60ba6c6465fd2060ccb/website%2Fdocs%2Fmlflow%2Finstallation.md) | 78 | 100 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/55c8b4f4-189a-42f1-8fdd-c9e74072aab9) | [:white_check_mark:](# "Successfully processed") |
| [website/docs/mlflow/introduction.md](https://github.com/microsoft/SynapseML/blob/42e0eaa90579b62d3277a60ba6c6465fd2060ccb/website%2Fdocs%2Fmlflow%2Fintroduction.md) | 88 | 100 | 71 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/3828919c-ed7b-4189-b1f5-e808fe628ca3) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

File: notebooks/features/hyperparameter_tuning/HyperOpt-SynapseML.ipynb [hunk 4]
```diff
@@ -49,7 +54,31 @@
...
+   "source": [
+    "from synapse.ml.core.platform import *\n",
+    "from pyspark.sql import SparkSession\n",
+    "\n",
+    "spark = SparkSession.builder.getOrCreate()\n",
+    "\n",
+    "if running_on_synapse_internal():\n",
+    "    experiment_name = \"hyperopt-synapseml\"\n",
+    "elif running_on_synapse():\n",
+    "    experiment_name = \"/Shared/hyperopt-synapseml\"\n",
```
Comment (serena-ruan) [inline notebooks/features/hyperparameter_tuning/HyperOpt-SynapseML.ipynb:75]:
```suggestion
    "    experiment_name = \"hyperopt-synapseml\"\n",
```
Let's use hyperopt-synapseml as experiment name directly, only databricks internally has a constraint on the experiment name has to be an absolute path.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/mlflow/installation.md](https://github.com/microsoft/SynapseML/blob/39e614b43f3acd677bf2a3b86c480b8b3bc0b556/website%2Fdocs%2Fmlflow%2Finstallation.md) | 78 | 100 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/fed7d76a-9b09-4ca2-b8c9-0bdff371b350) | [:white_check_mark:](# "Successfully processed") |
| [website/docs/mlflow/introduction.md](https://github.com/microsoft/SynapseML/blob/39e614b43f3acd677bf2a3b86c480b8b3bc0b556/website%2Fdocs%2Fmlflow%2Fintroduction.md) | 88 | 100 | 71 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/3723f40a-ed10-4e1e-84a2-fef4f5e6cb98) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

File: notebooks/features/hyperparameter_tuning/HyperOpt-SynapseML.ipynb [hunk 5]
```diff
@@ -49,7 +54,31 @@
...
+    "from pyspark.sql import SparkSession\n",
+    "\n",
+    "spark = SparkSession.builder.getOrCreate()\n",
+    "\n",
+    "if running_on_synapse_internal():\n",
+    "    experiment_name = \"hyperopt-synapseml\"\n",
+    "elif running_on_synapse():\n",
+    "    experiment_name = \"hyperopt-synapseml\"\n",
+    "    # from notebookutils.visualization import display # use this display on interactive notebook\n",
+    "    from synapse.ml.core.platform import (\n",
```
Comment (serena-ruan) [inline notebooks/features/hyperparameter_tuning/HyperOpt-SynapseML.ipynb:77]: Why is this 'display' needed for Synapse? Does SJD not include 'display' function?

**Comment by serena-ruan** [comment]
Good work @JessicaXYWang! Could you following this part in the PR description
image
and check markdown rendered correctly? We should make sure the website builds correctly each time we make changes to the website and this is a quick way to check locally, it starts localhost:3000.

File: website/docs/mlflow/installation.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+### Install mlflow on Databricks
+
+If you're using Databricks, install mlflow with this command:
+```
+# run this so that mlflow is installed on workers besides driver
+%pip install mlflow
+```
+
+### Install mlflow on Synapse
+mlflow is pre-installed on Synapse. To log model with mlflow, you'll need to create an Azure Machine Learning workspace and linking it with your Synapse workspace.
```
Comment (serena-ruan) [inline website/docs/mlflow/installation.md:24]:
```suggestion
MLflow is pre-installed on Synapse. To log model with MLflow, you'll need to create an Azure Machine Learning workspace and link it with your Synapse workspace.
```

**Review by mhamilton723 — CHANGES_REQUESTED** [review]
nit: would you be able to address the acrolinx required changes here?

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/mlflow/installation.md](https://github.com/microsoft/SynapseML/blob/1e7f29a1ebfa4602fa55e09dac379121853f10b3/website%2Fdocs%2Fmlflow%2Finstallation.md) | 100 | 100 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/575fb292-7bd5-4f2f-a291-c93f237a3e4d) | [:white_check_mark:](# "Successfully processed") |
| [website/docs/mlflow/introduction.md](https://github.com/microsoft/SynapseML/blob/1e7f29a1ebfa4602fa55e09dac379121853f10b3/website%2Fdocs%2Fmlflow%2Fintroduction.md) | 94 | 100 | 71 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/82245ac2-46aa-4953-9839-47305208ec7f) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Comment by JessicaXYWang** [comment]
> Good work @JessicaXYWang! Could you follow this part in the PR description image and check markdown rendered correctly? We should make sure the website builds correctly each time we make changes to the website and this is a quick way to check locally, it starts localhost:3000.

Thanks. I checked and it looks good. 

When I try to start it locally, I need to do 

1. nbconvert notebooks before starting it otherwise it raises errors.  - is this expected?

2. use `yarnpkg run start` rather than `yarn run start` due to a hadoop YARN and Yarn package command conflicts. Let me know if you want to update the website starting docs.

**Comment by serena-ruan** [comment]
> > Good work @JessicaXYWang! Could you follow this part in the PR description image and check markdown rendered correctly? We should make sure the website builds correctly each time we make changes to the website and this is a quick way to check locally, it starts localhost:3000.
> 
> Thanks. I checked and it looks good.
> 
> When I try to start it locally, I need to do
> 
> 1. nbconvert notebooks before starting it otherwise it raises errors.  - is this expected?
> 2. use `yarnpkg run start` rather than `yarn run start` due to a hadoop YARN and Yarn package command conflicts. Let me know if you want to update the website starting docs.

1. Ye it's expected.
2. Well I personally don't install hadoop YARN locally so my 'yarn run start' works fine :D

**Review by serena-ruan — APPROVED** [review]
Thanks!

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**